api/v1/storageprofile_types.go

```diff
@@ -0,0 +1,103 @@
+type StorageProfileSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="DeviceClass is immutable"
+	// +kubebuilder:validation:MaxLength=512
+	// DeviceClass is the deviceclass name.
+	DeviceClass string `json:"deviceClass"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SharedFilesystemConfiguration is immutable"
+	// configurations to use for cephfilesystem.
+	SharedFilesystemConfiguration SharedFilesystemConfigurationSpec `json:"sharedFilesystemConfiguration,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="BlockPoolConfiguration is immutable"
+	// configurations to use for  profile specific blockpool.
+	BlockPoolConfiguration BlockPoolConfigurationSpec `json:"blockPoolConfiguration,omitempty"`
+}
+
+// StorageProfileStatus defines the observed state of StorageProfile
+type StorageProfileStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Phase describes the Phase of StorageProfile
+	// This is used by OLM UI to provide status information
+	// to the user
+	Phase StorageProfilePhase `json:"phase,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// StorageProfile is the Schema for the storageprofiles API
+type StorageProfile struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StorageProfileSpec   `json:"spec,omitempty"`
+	Status StorageProfileStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// StorageProfileList contains a list of StorageProfile
+type StorageProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StorageProfile `json:"items"`
+}
+
+// StorageProfilePhase stores a StorageProfile reconciliation phase
+type StorageProfilePhase string
+
+const (
+	StorageProfilePhaseFailed StorageProfilePhase = "Failed"
+	StorageProfilePhaseInit   StorageProfilePhase = "Initializing"
+	StorageProfilePhaseReady  StorageProfilePhase = "Ready"
+)
+
+// StorageProfileMessage stores a StorageProfile reconciliation message
+type StorageProfileMessage string
+
+const (
+	StorageProfilePhaseReadyMessage StorageProfileMessage = "StorageProfile reconcile completed successfully"
+	StorageProfileNotFound          StorageProfileMessage = "no storage profile definition found for storage profile"
+	StorageProfileListError         StorageProfileMessage = "Unable to list StorageProfile objects"
```

Comment:
- if this is being used as an error it shouldn't be of type `StorageProfileMessage`
- it should be a error value and `errors.Is` can be used to compare the values, something like
``` go
import (
  "errors"
)

var errList = errors.New("Listing")

// some sample func
func do() {
  err := func() error { return errList }()
  if errors.Is(err, errList) { // do what we want for this specific error }
}
```